akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -499,16 +516,82 @@ class Http2ServerSpec extends AkkaSpec with WithInPendingUntilFixed with Eventua
+        emitResponse(1, tooLargeResponse)
+
+        expectHeaderBlock(1, endStream = false)
+        sendWINDOW_UPDATE(1, maxSize * 3) // make sure we can receive such large response, on this stream
+        // FIXME we should not need to send this window update, the stream already has enough window I think, see above ^^^
+        // we receive the DATA in 2 parts, since the ByteString does not fit in a single frame
+        val d1 = expectDATA(1, endStream = false, numBytes = maxSize)
+        val d2 = expectDATA(1, endStream = true, numBytes = maxSize)
+        d1.toList should have length maxSize
+        d2.toList should have length maxSize
+        (d1 ++ d2) should ===(theTooLargeByteString) // makes sure we received the parts in the right order
+      }
+
+      // http://httpwg.org/specs/rfc7540.html#rfc.section.4.2
+      "receive SETTINGS_MAX_FRAME_SIZE with too small value (spec 4.2)" in new TestSetup with RequestResponseProbes {
+        def setMaxFrameSize(size: Int) = SettingsFrame(Setting(SettingIdentifier.SETTINGS_MAX_FRAME_SIZE, size) :: Nil)
 
-      "received SETTINGS_MAX_CONCURRENT_STREAMS" in pending
+        /*
+          An endpoint MUST send an error code of FRAME_SIZE_ERROR if a frame exceeds the size defined in SETTINGS_MAX_FRAME_SIZE, 
+          exceeds any limit defined for the frame type, or is too small to contain mandatory frame data. 
+         */
+
+        sendFrame(setMaxFrameSize(8))
 
-      "received SETTINGS_HEADER_TABLE_SIZE" in {
+        val (lastStreamId, errorCode) = expectGOAWAY()
+        errorCode should ===(ErrorCode.FRAME_SIZE_ERROR)
+      }
+
+      "receive SETTINGS_MAX_CONCURRENT_STREAMS" in pending
+
+      "receive SETTINGS_HEADER_TABLE_SIZE" in new TestSetup with RequestResponseProbes {
         // if the sender of the new size wants to shrink its decoding table, the encoding table on
         // our side needs to be shrunk *before* sending the SETTINGS ACK. So a mechanism needs to be
         // found that prevents race-conditions in the encoder between sending out an encoded message
         // which still uses the old table size and sending the SETTINGS ACK.
-        pending
+
+        def setHeaderTableSize(size: Int) = SettingsFrame(Setting(SettingIdentifier.SETTINGS_HEADER_TABLE_SIZE, size) :: Nil)
+
+        private val maxSize = Math.pow(2, 15).toInt // 32768, valid value (between 2^14 and 2^14 - 1)
+        sendFrame(setHeaderTableSize(maxSize))
+
+        expectSettingsAck()
+      }
+
+      "received disable PUSH_PROMISE by received SETTINGS_ENABLE_PUSH (invalid 6.5)" in new TestSetup with RequestResponseProbes {
+        def setAllowPushPromise(value: Boolean) = SettingsFrame(Setting(SettingIdentifier.SETTINGS_ENABLE_PUSH, if (value) 1 else 0) :: Nil)
+
+        sendFrame(setAllowPushPromise(false))
+
+        expectSettingsAck()
+
+        // FIXME should send a PUSH_PROMISE from server side now, it should fail since it's not allowed
```

Comment:
Actually, it shouldn't fail but the server push should just be ignored and maybe issue a warning on a low logging level. Maybe let's keep this test pending until it can check anything.